
syntax = "proto3";

option java_multiple_files = true;
option optimize_for = SPEED;
package reader;

// example of loop as a grpc C++ service 

service ReaderService {
   // basic grpc call with blocking
   rpc GetRecord(ReaderRequest) returns (ReaderResponse);
   rpc GetRecordString(ReaderRequest) returns (ReaderResponseString);
}

service PingService {
   rpc Ping(PingRequest) returns (PingResponse);
}

message PingRequest {
    string message = 1;
}

message PingResponse {
    string message = 1;
}

// find all record in CSV having the field key == value

message ReaderRequest {
    string key = 1;
    string value = 2; // TODO, add string option later
}

message ReaderResponse {
    repeated Record records = 1;
    int64 num_record = 2;
}


//message ReaderValue {
//    oneof values {
//        int64 int_value = 1;
//        string string_value = 2;
//        //TODO: add double later
//    }
//}

message Record {
    repeated Field fields = 1;
}

message Field {
    string key = 1;
    string value = 2;
}

message ReaderResponseString {
    string data = 1;
}